tests/integration/backward_compatible/serialization/compact_test.py

```diff
@@ -784,6 +784,24 @@ def listener(_):
         self.assertTrueEventually(lambda: self.assertEqual(1, counter.get()))
 
 
+class CompactWithNestedFieldTest(CompactTestBase):
```

Comment:
I feel like this can easily be a unit test. Can you move it to `tests/unit/serialization/compact_test.py` and modify it such that it doesn't need a cluster? You can take a look at the other tests in that directory for reference. Just construct a serialization service with appropriate configuration, call to_data on it and expect it to throw the exception

Reply:
Converted it to a unit test at https://github.com/hazelcast/hazelcast-python-client/pull/574/commits/3e052df6aaaf866e1f977a76832ca4c1577e85f4. I had to a custom schema registration function to simulate the `put()` function of `MapProxy`. Node.js Client has similar util functions too. See at [CompactUtil.js#L1483](https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/test/integration/backward_compatible/parallel/serialization/compact/CompactUtil.js#L1483).